### DISCUSSION TITLE
app/etc/local.xml compromised

### DISCUSSION THREAD

**Discussion by luigifab** [OpenMage/magento-lts#1882] [Questions & Answers]
Due to a misconfiguration of the server, we have found that some websites expose their _app/etc/local.xml_.
What are the consequences? 💣

**Comment by kiatng** [comment]
Bad actor may gain access to the DB, if that happens, the potential damage is exponentially propotional to the worth of the data.

**Comment by tmotyl** [comment]
Are you in EU? Then consult your lawyer regarding country specific rules about gdpr. In Poland you have 48h to notify in a formal way the administrator of the personal data of the system (usually the merchant).

**Reply by luigifab** [reply]
Yes, all owners are informed.

**Comment by Flyingmana** [comment]
This is exactly the right question to start a risk analyses.
You need to check everything which is noted in the local.xml regarding login and api credentials, and also the backend url.
Use this as base, to check all of them if they would be accessable from the outside, and if the credentials were usable for this.
You need to actually check/try this. "It should not be accessable" is not good enough for the analyses.

Usually the biggest risk here is the backend url, although there are rare cases of misconfigured server setups, which also expose mysql or redis to the web.
Check the Backend logs for unusual logins or activity for since the local.xml was viewable.

**Reply by luigifab** [reply]
This critical fail of the life concern only tests websites.
The database of theses websites use a copy of production databases without customers and orders/invoices/shipments/refunds...
The sql and redis server are not accessible from internet.

We have changed all users/passwords, and all backend url.
But the local.xml also expose the **cypt key**, not sure what to think about it.

**Reply by Flyingmana** [reply]
ahh right, crypt key, you probably should check where it is used for. By default I think only encrypted config settings (so api keys and passwords) are encrypted with it. But that is only relevant if they also got access to the database. Or it leaks somewhen in the future(or past).
When changing it, keep in mind you need to also migrate these settings.

**Comment by elidrissidev** [comment]
Oh boy! doesn't Magento actually show a warning in the admin panel when it's misconfigured? Pretty sure I saw it before. It happened to us before, some bots were crawling the website looking for configuration files but luckily we had everything configured properly.

**Reply by kiatng** [reply]
Could be here 

https://github.com/OpenMage/magento-lts/blob/1.9.4.x/app/code/core/Mage/Adminhtml/Block/Notification/Security.php

and reference here

https://github.com/OpenMage/magento-lts/blob/c8a627fa7338244479d78600f098d8d3b7a76a49/app/design/adminhtml/default/default/layout/main.xml#L121

and this [here](https://www.yireo.com/tutorials/magento/magento-administration/1322-client-denied-by-server-configuration-appetclocalxml) show how it can be disabled.

**Reply by elidrissidev** [reply]
@luigifab can you confirm if the affected installations had this notification visible in the admin?

**Reply by luigifab** [reply]
I checked, notification is visible.

**Comment by addison74** [comment]
Is it an Nginx webserver, isn't it? It happens if you don't configure it properly. 

1. If the database server port is public and it can be accessed from outside of the network you are in trouble. This is newbie issue MySQL server should run always on localhost. You can expose the port only changed as number for a limited time.
2. Sometimes the authentication information from database are the same with the ones from SSH server. If the SSH is running on the default port it is piece of cake for anyone who knows Linux commands to access the system. 
3. If a database management script is installed (e.g. phpMyAdmin) and someone knows the path he will have access to your databases for that user.

Only a stupid hacker reveals his actions like deleting files or changing files. Those who are smart continue staying undercover and grabbing information. If you have root access to that machine you can check the logs. You can check webserver logs who accessed local.xml. I would go further in checking what files were created, accessed, modified in /home/<user>.

**Reply by luigifab** [reply]
True, it's Nginx. For all your points, for our cases, response is no.
